### PR TITLE
Fix different build behavior for single-user and multi-user nix, disable update notifier

### PR DESCRIPTION
**Pull request by zhenyavinogradov**

### CODE REVIEW

**Comment by balsoft** [comment]
Alternatively, you can set `HOME` to some writable place (like `/build`)

**Comment by zhenyavinogradov** [comment]
@balsoft It works too, but it's safer to forbid writing to $HOME at all to be sure npm does not preserve any unexpected state between invocations

File: default.nix [hunk 1]
```diff
@@ -141,6 +142,15 @@ in rec {
...
       npmFlags = npmFlagsNpm;
       buildCommand = ''
+        # Inside nix-build sandbox $HOME points to a non-existing
+        # directory, but npm may try to create this directory (e.g.
+        # when you run `npm install` or `npm prune`) and will succeed
+        # if you have a single-user nix installation (because / is
+        # writable in this case), causing different behavior for
+        # single-user and multi-user nix. Create read-only $HOME to
+        # prevent it
+        mkdir -p --mode=a-w "$HOME"
```
Comment (kirelagin) [inline default.nix:152]: Wait, what? Are you saying that single-user Nix with sandboxing enabled has writeable `/`? Isn’t that a bug? Why would anyone want this?

**Comment by kirelagin** [comment]
Please, write `Fixes #<issue>` when opening a PR that fixes an issue.

**Comment by kirelagin** [comment]
And do not write it in commit messages, because... um... in general it is weird to use GitHub-specific things in pure git.

File: default.nix (hunk 1 above)
Comment (zhenyavinogradov) [reply default.nix:152]: I think on multi-user nix `/` is protected by having a different owner, so maybe they didn't consider it important enough to implement another logic just for a single-user mode

**Comment by zhenyavinogradov** [comment]
Argh damn

**Comment by balsoft** [comment]
FTR the error message is 
```
  mkdir: cannot create directory '/homeless-shelter': Permission denied
```

**Comment by kirelagin** [comment]
Just add `|| true`? 🤷

**Comment by zhenyavinogradov** [comment]
https://github.com/serokell/nix-npm-buildpackage/pull/20 fixes it, thank you @balsoft

**Pull request merged**